packages/components/inputs/toggle-input/src/toggle-input.styles.ts

```diff
@@ -0,0 +1,209 @@
+// TODO: @redesign cleanup
+import styled from '@emotion/styled';
+import { css } from '@emotion/react';
+import {
+  designTokens,
+  type ThemeName,
+} from '@commercetools-uikit/design-system';
+import type { TToggleInputProps } from './toggle-input';
+
+type NewThemeProps = {
+  isNewTheme: boolean;
+  theme: ThemeName;
+};
+
+type TStyledLabelProps = Pick<TToggleInputProps, 'isDisabled' | 'size'> &
+  NewThemeProps;
+type TStyledSpanProps = Pick<TToggleInputProps, 'size'> & NewThemeProps;
+
+const track = {
+  test: {
```

Comment:
Hmm not sure if hardcoding the theme here should be a concern of this component.

I would map the values but use the `themedValue()` function to render one value or another.

Reply:
Thanks, TBH that was my starting point but I thought you wouldn't want that many `themedValue` calls. 
Wondering if  this notation `thumb(props.themedValue)[props.size].hoverAreaWidth` is too messy...

Reply:
> Wondering if this notation `thumb(props.themedValue)[props.size].hoverAreaWidth` is too messy...

Up to you to judge. 🤗

In general you should ask yourself if someone would be comfortable with this and understand the logic when reading it (even without a context). That often times includes you as well as the author (for example, _in a week time, if you read this, do you know what it does?_).

Not saying you need to change this, just suggesting how you can think about it and make a decision.

Reply:
Right, thanks 🙂 Well, I suggested this so I'm ok with that 😅
So the question was rather if anyone else finds this hard to read.